LibDebug: Propagate errors around LineProgram

BenWiederhake opened the pull request:
Found while playing Fixme-Roulette.

This was pleasantly easy because both caller and callee already were set up for proper error propagation.

LucasChollet commented on Userland/Libraries/LibDebug/Dwarf/LineProgram.h:
```diff
@@ -108,7 +109,7 @@ class LineProgram {
 
 public:
-    explicit LineProgram(DwarfInfo& dwarf_info, SeekableStream& stream);
+    static ErrorOr<NonnullOwnPtr<LineProgram>> try_create(DwarfInfo& dwarf_info, SeekableStream& stream);
```
https://github.com/SerenityOS/serenity/blob/master/Documentation/Patterns.md#fallible-constructors

Please use `create` instead of `try_create`

BenWiederhake commented:
Changes since last push:
- Renamed `try_create` to `create` in accordance with the naming convention (Thanks @LucasChollet!)
- Rebased for convenience

konradekk reviewed (approved):
does not look too bad…! 😉
